browser/ui/webui/settings/brave_importer_observer.h

```diff
@@ -0,0 +1,47 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_BROWSER_UI_WEBUI_SETTINGS_BRAVE_IMPORTER_OBSERVER_H_
+#define BRAVE_BROWSER_UI_WEBUI_SETTINGS_BRAVE_IMPORTER_OBSERVER_H_
+
+#include "base/callback.h"
+#include "base/memory/raw_ptr.h"
+#include "base/values.h"
+#include "chrome/browser/importer/importer_progress_observer.h"
+
+class ExternalProcessImporterHost;
+
+class BraveImporterObserver : public importer::ImporterProgressObserver {
+ public:
+  using ReportProgressCallback = base::RepeatingCallback<
+      void(const importer::SourceProfile& source_profile, const base::Value&)>;
+
+  explicit BraveImporterObserver(ExternalProcessImporterHost* host,
+                                 const importer::SourceProfile& source_profile,
+                                 uint16_t imported_items,
+                                 ReportProgressCallback callback);
+  ~BraveImporterObserver() override;
+
+  void ImportStarted() override;
+  void ImportItemStarted(importer::ImportItem item) override;
+  void ImportItemEnded(importer::ImportItem item) override;
+  void ImportEnded() override;
+
+ private:
+  FRIEND_TEST_ALL_PREFIXES(BraveImporterObserverUnitTest, ImportEvents);
+
+  ExternalProcessImporterHost* GetImporterHostForTesting();
```

Comment:
nit: Please add blank line between method and vars.

Reply:
done, thanks